controllers/group_controller.go

```diff
@@ -159,3 +159,36 @@ func GetUsersOfGroups(ctx *gin.Context) {
 	}
 	SendResponse(ctx, 200, groupUsers, "Fetched group users", nil)
 }
+
+func GetUsersWithoutGroupAccess(ctx *gin.Context) {
+	userID, err := utils.FetchUserIDFromCtx(ctx)
+	if err != nil {
+		SendResponse(ctx, 401, nil, "Unauthorized", errors.New("unauthorized"))
+		return
+	}
+
+	groupIDStr := ctx.Param("groupId")
+	groupID, err := uuid.Parse(groupIDStr)
+	if err != nil {
+		SendResponse(ctx, 400, nil, "Invalid group id", nil)
+		return
+	}
+	// Check user is authorized to see members of the group
+	isMember, err := service.CheckUserMemberOfGroup(ctx, userID, groupID)
+	if !isMember {
+		SendResponse(ctx, 401, nil, "", errors.New("unauthorized for group view"))
+		return
+	}
+	if err != nil {
+		SendResponse(ctx, 500, nil, "Failed to fetch Group members", errors.New("failed to fetch group members"))
+		return
+	}
+
+	users, err := service.GetUsersWithoutGroupAccess(ctx, groupID)
+
+	if err != nil {
```

Comment:
We should change the way we call SendResponse. We have made this mistake in a lot of place
if the request is a failure, we should pass an error to the SendResponse func. 

Something like 
`SendResponse(ctx, 500, nil, "", errors.New("failed to fetch users without group access"))`

We do this because `SendResponse` marks an api call as failure only if the error value is not null